CodeforAustralia/cm-seechange-survey-blueprint
# Initial Nextjs & eslint setup

## Overview of the Pull Request:
This PR introduces the following changes to repo:
* add `.nvmrc` file to setup nvm environment
* Initial setup of a Next.js app
* default eslint config for app (see `.eslintrc.json`)
* configured Husky with a pre-commit git hook to run eslint (see `.husky/pre-commit`)
* add `docs/00_initial_project_setup.md` to document project setup for reference
* update `README.md` with instructions for contributors to setup local dev environment


There are many other files included in this commit; they were all generated as part of the initial default setup of a Next.js app (see `docs/00_initial_project_setup.md` for details). They can probably be safely ignored and do not require review at this stage. Things to note with a default Next.js app setup is this:
* Next.js ships with [TypeScript](https://nextjs.org/docs/app/building-your-application/configuring/typescript), [ESLint](https://nextjs.org/docs/app/building-your-application/configuring/eslint), and [Tailwind CSS](https://nextjs.org/docs/app/building-your-application/styling/tailwind-css) configuration by default.
* this setup places all of the application's code in `src/` directory.
see [this doc](https://nextjs.org/docs/getting-started/installation) for more details on a default Next.js app install.

## Things to focus for review in this PR are:
* `.eslintrc.json` - the eslint config
* `.husky/pre-commit` - the pre-commit git hook to run eslint before each git commit
* `docs/00_initial_project_setup.md` - the reference doc for setting up new Next.js app
* `README.md` - the `Getting Started for Local Development` section of doc

## link to Trello card or Github issue
https://trello.com/c/tPsbwter

## How to Test this PR?
* Review the `Getting Started for Local Development` section of `README.md`
  * follow the instruction; confirm that the instructions contain sufficient info to set up a working dev environment
  * confirm that Next.js app runs in dev environment
* Test to ensure the eslint config catches linting errors as expected
  * introduce a linting error in `src/` folder, then run `npm run lint` to confirm that eslint reports an error
* Test that pre-commit git hook works
  * introduce a linting error in `src/` folder, then run `git commit` ; confirm that the git operation fails due to eslint reporting an error
* review this doc: `docs/00_initial_project_setup.md` - confirm it is accurate, readable, and makes sense.
 

## Pull Request Checklist:

Make sure the following checkboxes`[ ]` are checked with `x` before sending your PR -- thank you!

- ~Have you checked That code is well formatted? Did `npx prettier --check .` return no warnings/errors?~
